fishtank/src/cluster.test.ts

```diff
@@ -5,16 +5,126 @@ import { Docker } from './backend'
+
+      await cluster.init()
+
+      expect(createNetwork).toHaveBeenCalledWith('my-test-cluster', {
+        attachable: true,
+        internal: true,
+        labels: { 'fishtank.cluster': 'my-test-cluster' },
+      })
+      expect(runDetached).toHaveBeenCalledWith('ironfish:latest', {
+        args: ['start'],
+        name: 'my-test-cluster_bootstrap',
+        networks: ['my-test-cluster'],
+        hostname: 'bootstrap',
+        labels: { 'fishtank.cluster': 'my-test-cluster', 'fishtank.node.role': 'bootstrap' },
+      })
+    })
+
+    it('only creates the network if bootstrap was not requested', async () => {
+      const backend = new Docker()
+      const cluster = new Cluster({ name: 'my-test-cluster', backend })
+
+      const createNetwork = jest
+        .spyOn(backend, 'createNetwork')
+        .mockReturnValue(Promise.resolve())
+      const runDetached = jest.spyOn(backend, 'runDetached').mockReturnValue(Promise.resolve())
+
+      await cluster.init({ bootstrap: false })
+
+      expect(createNetwork).toHaveBeenCalledWith('my-test-cluster', {
+        attachable: true,
+        internal: true,
+        labels: { 'fishtank.cluster': 'my-test-cluster' },
+      })
+      expect(runDetached).not.toHaveBeenCalled()
+    })
+  })
+
+  describe('bootstrap', () => {
+    it('launches a bootstrap node', async () => {
+      const backend = new Docker()
+      const cluster = new Cluster({ name: 'my-test-cluster', backend })
+
+      const runDetached = jest.spyOn(backend, 'runDetached').mockReturnValue(Promise.resolve())
+
+      await cluster.bootstrap()
+
+      expect(runDetached).toHaveBeenCalledWith('ironfish:latest', {
+        args: ['start'],
+        name: 'my-test-cluster_bootstrap',
+        networks: ['my-test-cluster'],
+        hostname: 'bootstrap',
+        labels: { 'fishtank.cluster': 'my-test-cluster', 'fishtank.node.role': 'bootstrap' },
+      })
+    })
+
+    it('launches a bootstrap node with the given name', async () => {
+      const backend = new Docker()
+      const cluster = new Cluster({ name: 'my-test-cluster', backend })
+
+      const runDetached = jest.spyOn(backend, 'runDetached').mockReturnValue(Promise.resolve())
+
+      await cluster.bootstrap({ nodeName: 'my-bootstrap-node' })
+
+      expect(runDetached).toHaveBeenCalledWith('ironfish:latest', {
+        args: ['start'],
+        name: 'my-test-cluster_my-bootstrap-node',
+        networks: ['my-test-cluster'],
+        hostname: 'my-bootstrap-node',
+        labels: { 'fishtank.cluster': 'my-test-cluster', 'fishtank.node.role': 'bootstrap' },
+      })
+    })
+
+    it('launches a bootstrap node with the given image', async () => {
+      const backend = new Docker()
+      const cluster = new Cluster({ name: 'my-test-cluster', backend })
+
+      const runDetached = jest.spyOn(backend, 'runDetached').mockReturnValue(Promise.resolve())
+
+      await cluster.bootstrap({ nodeImage: 'some-image' })
```

Comment:
verify response

Reply:
You mean the return value? This is `void`, how can I verify it? Or do you mean something else?

Reply:
I meant to comment it for `spawn()`, made a new comment.